### PR TITLE
RadioGroup accessibility + specs

### PR DESCRIPTION
**Pull request by giladgrush**
Apart from adding information relevant for accessibility and keyboard handling, I added specs which  should help us think about a new implementation pattern according to what we learned from the SelectionList

### CODE REVIEW

File: specs/radiogroup.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+Support a similar to native radiogroup implementations. Please see SelectionList implementation to view the new pattern.
+
+## Properties
+
+| Name | Type | Default | Required | Description |
+| -- | -- | -- | -- | -- |
+| value | ?R&D? | false | no | The value chosen in the checkbox |
+| onChange | ?R&D? | NOOP | no | Triggered by changing a radio button state |
+| children | Array<Node> | null | no | children | radio buttons and/or other nodes which will just be rendered |
+| dataSource | Array\<Object> | [] | no | There are a few options accepted as a datasource (see below for explanation) |
```
Comment (JoMarton) [inline specs/radiogroup.md]: Do we mean `Array<dataSchema>` ? also what explanation is below?

File: specs/radiogroup.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+| dataSchema | object | { id: 'id', displayText: 'displayText' } | no | Maps the object properties to the relevant properties required by the ItemRenderer |
```
Comment (JoMarton) [inline specs/radiogroup.md]: should dataSchema indicate if item is selected by default? disabled? name? in general the whole interface of a RadioButton

File: specs/radiogroup.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+* Text content attributes relevant for the radiogroup - aria-label, aria-labelledby, aria-describedby.
+* aria-checked is set to true on the checked radio button and set to false on the unchecked radio buttons.
+* aria-activedescendant="ID_REF" - Applied to the element with the radiogroup role, tells assistive technologies which of the options, if any, is visually indicated as having keyboard focus. idref, refers to the ID of the focused option in the radio group. When navigation keys, such as Down Arrow, are pressed, the JavaScript changes the value.
+
+### Focus
+
+* Upon receiving focus:
+  * If a radio button is checked, focus is set to the checked button
+  * If no button is checked, focus is set to the first radio button
+* Set tabindex="-1" on all radio buttons which are unchecked (apart from the first one if all are unchecked), otherwise set tabindex="0"
```
Comment (JoMarton) [inline specs/radiogroup.md]: I see aria-activedescendant in aria attributes... should we combine it with tabIndex roving?

File: specs/radiogroup.md (hunk 1 above)
Comment (JoMarton) [inline specs/radiogroup.md]: what happens when dataSource and children are supplied together?

File: specs/radiogroup.md (hunk 2 above)
Comment (giladgrush) [inline specs/radiogroup.md]: It's possible. This object is used to map the information coming from the datasource to the itemrenderer (default or custom). You can decide on anything you want.

File: specs/radiogroup.md (hunk 3 above)
Comment (giladgrush) [inline specs/radiogroup.md]: Just use roving tabindex.

**Pull request merged**